src/www/model/events.ts

```diff
@@ -55,9 +49,13 @@ export class ServerReconnecting implements OutlineEvent {
 }
 
 // Simple publisher-subscriber queue.
+interface PublicServerConstructible {
+  new(server: Server): OutlineEvent;
+  server?: Server;
+}
 export class EventQueue {
   private queuedEvents: OutlineEvent[] = [];
-  private listenersByEventType = new Map<OutlineEvent, OutlineEventListener[]>();
+  private listenersByEventType = new Map<PublicServerConstructible, OutlineEventListener[]>();
```

Comment:
We don't care about how the event is constructed. This type doesn't belong here.